src/classes/RD2_ValidationService.cls

```diff
@@ -67,33 +67,37 @@ public with sharing class RD2_ValidationService {
 
         for (npe03__Recurring_Donation__c rd : rds) {
             if (rd.npe03__Organization__c == null && rd.npe03__Contact__c == null) {
-            rd.addError(System.Label.RD_DonorIsRequired);
-            return;
-        }
+                rd.addError(System.Label.RD_DonorIsRequired);
+                return;
+            }
             validate(rd, accountsById.get(rd.npe03__Organization__c));
         }
     }
 
     /***
     * @description Validates missing account or contact data on Recurring Donation
     * @param rd the Recurring Donation to validate
-    * @param accountById is the associated account for the Recurring Donation
+    * @param acct is the associated account for the Recurring Donation
     */
-    public void validate(npe03__Recurring_Donation__c rd, Account accountById) {
+    public void validate(npe03__Recurring_Donation__c rd, Account acct) {
 
-        if (accountbyId != null) {
-            if (accountById.npe01__SYSTEMIsIndividual__c && rd.npe03__Contact__c == null) {
-                rd.addError(System.Label.RD_DonorIsRequired);
+        if (acct == null) {
+            return;
+        }
+
+        if (acct.npe01__SYSTEMIsIndividual__c) {
+            if (rd.npe03__Contact__c == null) {
+                rd.addError(System.Label.RD_ContactMustBelongToAccount);
             }
 
-            else if (accountById.npe01__SYSTEMIsIndividual__c && (!validateContact(rd, accountById.Contacts) || accountById.Contacts.size() == 0)) {
+            else if (!validateContact(rd, acct.Contacts) || acct.Contacts.size() == 0) {
```

Comment:
Our style guide has the else surrounded by brackets: `} else if () {`

Reply:
Good catch @force2b .